src/traces/candlestick/attributes.js

```diff
@@ -0,0 +1,49 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Lib = require('../../lib');
+var OHLCattrs = require('../ohlc/attributes');
+var boxAttrs = require('../box/attributes');
+
+var directionAttrs = {
+    visible: {
```

Comment:
I think the main reason I allowed the 2 visibilities to be controlled was because that was the easiest way for me to make sure users could edit specific attributes of each trace more easily (color/name etc). I think I wasn't sure how'd these would be used/styled and what attributes would be important. So I'd say if we can't think of a use case of allowing separate control of the visibilities here (I can't think of one) then we don't have to make this completely parallel to the python wrapper.